sqlhooks.go

```diff
@@ -102,6 +102,13 @@ func (conn *Conn) PrepareContext(ctx context.Context, query string) (driver.Stmt
 func (conn *Conn) Prepare(query string) (driver.Stmt, error) { return conn.Conn.Prepare(query) }
 func (conn *Conn) Close() error                              { return conn.Conn.Close() }
 func (conn *Conn) Begin() (driver.Tx, error)                 { return conn.Conn.Begin() }
+func (conn *Conn) Ping(ctx context.Context) error {
+	p, ok := conn.Conn.(driver.Pinger)
+	if !ok {
+		return driver.ErrSkip
```

Comment:
ErrSkip is not documented as something supported by Ping. Reading the documentation of driver this error is only allowed to be returned when explicitly documented as supported for that function.

To be honest, to support Ping here properly is quite a bit of effort. However, its pretty bad that sqlhooks doesn't implement Ping since pretty much everyone should be supporting it.

So either we do the hard work of correctly supporting Ping, or we document a contract that Pinger has to be supported by whatever we wrap?

Reply:
> ErrSkip is not documented as something supported by Ping. Reading the documentation of driver this error is only allowed to be returned when explicitly documented as supported for that function.

As per the docs:
> // ErrSkip may be returned by some optional interfaces' methods to
// indicate at runtime that the fast path is unavailable and the sql
// package should continue as if the optional interface was not
// implemented.

What I understand is that the optional interface `Ping` implementation should return `ErrSkip` when not implemented by the underlying driver. 

I'd love to understand better what you mean @keegancsmith .

Also, assuming most drivers already implement `Ping`, what would be the downside of keeping this change?

Reply:
From https://pkg.go.dev/database/sql/driver@go1.22.4#pkg-variables
> ErrSkip may be returned by some optional interfaces' methods to indicate at runtime that the fast path is unavailable and the sql package should continue as if the optional interface was not implemented. ErrSkip is only supported where explicitly documented.

Note the mention of explicitly documented. Then look at the documentation of Pinger and it is not explicitly documented unlike some other optional interfaces.

The downside of keeping this change is the call sites of Ping inside of driver will fail rather than handling ErrSkip for drivers that do not support Pinger. I took a quick look at the code and think non-pinger supported drivers will be broken.

However, I think this project should support this interface given in practice I assume all drivers support it.

Reply:
@wburningham are you aware of any sql driver that does not implement the `Pinger` interface so that we can test the `ErrSkip` behavior and see how bad the damage will be?

Reply:
is that something you can try and share the outcome?